File: fathersDay/Dsa/Day8 (1979. Find Greatest Common Divisor of Array)/Solution.py
from math import gcd

class Solution:
    def FindGCD(self, nums: list[int]) -> int:
        return gcd(min(nums), max(nums))
    
# Example usage
solution = Solution()
arr = [2, 5, 6, 9, 10]
print(solution.FindGCD(arr))  # Output: 2

File: fathersDay/Dsa/Day8 (1979. Find Greatest Common Divisor of Array)/Solution.java
class Solution {
    // Function to calculate GCD using Euclidean algorithm
    private int calculateGCD(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public int findGCD(int[] nums) {
        // Find the minimum and maximum elements in the array
        int minVal = nums[0];
        int maxVal = nums[0];

        for (int num : nums) {
            if (num < minVal) {
                minVal = num;
            }
            if (num > maxVal) {
                maxVal = num;
            }
        }
        
        // Return the GCD of the minimum and maximum elements
        return calculateGCD(minVal, maxVal);
    }
}